src/nsidc/metgen/metgen.py

```diff
@@ -468,18 +476,21 @@ def grouped_granule_files(configuration: config.Config) -> list[tuple]:
     file_list = [p for p in Path(configuration.data_dir).glob("*")]
 
     if configuration.granule_regex:
-        granule_name_fragments = set(
-            re.search(configuration.granule_regex, file.name).group("granuleid")
-            for file in file_list
+        # file_list -> matches -> filtered matches -> granuleids -> set
+        pipeline = rcompose(
+            partial(re.match, configuration.granule_regex),
```

Comment:
I used `re.search` because the docs tell me `search` matches anywhere in the string, whereas `re.match` matches at the beginning of the string. Am I misinterpreting the description?

Reply:
No, you're right, and I forgot to revert this. I switched it when I thought I needed a more complicated regex.